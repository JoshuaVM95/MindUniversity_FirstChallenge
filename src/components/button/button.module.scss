.button {
	border: none;
	color: #fff;
	cursor: pointer;

	&:hover {
		filter: brightness(1.5);
	}

	&:active {
		filter: saturate(1.5);
	}

	&:disabled {
		opacity: 0.8;
		cursor: not-allowed;
	}

	&.small {
		width: 4rem;
		height: 1rem;
		font-size: 1.2rem;
	}

	&.medium {
		font-size: 1.8rem;
		width: 16rem;
		height: 4rem;
	}

	&.large {
		font-size: 1.8rem;
		width: 21rem;
		height: 5rem;
	}
}
