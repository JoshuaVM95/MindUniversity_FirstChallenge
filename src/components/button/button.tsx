import React from 'react'
import styles from './button.module.scss'

interface ButtonProps {
	text: string
	size: 'small' | 'medium' | 'large'
	color: string
	onClick(event: React.MouseEvent): void
	disabled?: boolean
}

export const Button = ({ text, size, color, onClick, disabled }: ButtonProps): React.ReactElement => {
	return (
		<button
			className={`${styles.button} ${styles[size]}`}
			style={{ backgroundColor: color }}
			onClick={onClick}
			disabled={disabled}
		>
			{text}
		</button>
	)
}
