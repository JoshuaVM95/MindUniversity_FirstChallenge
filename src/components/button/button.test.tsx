import React from "react";
import { Button } from "./button";
import { render, fireEvent, getByText } from "@testing-library/react";

describe("Button tests", () => {
  const click = jest.fn();
  const renderButton = () => {
    return render(
      <Button text="Test Btn" size="small" onClick={click} color="blue" />
    );
  };
  it("renders correctly", () => {
    expect(renderButton().asFragment()).toMatchSnapshot();
  });
  it("renders the text prop correctly", () => {
    const button = renderButton().asFragment();
    expect(button).toHaveTextContent("Test Btn");
  });
  it("triggers click", () => {
    const button = renderButton().container;
    fireEvent.click(getByText(button, "Test Btn"));
    expect(click).toHaveBeenCalled();
  });
});
