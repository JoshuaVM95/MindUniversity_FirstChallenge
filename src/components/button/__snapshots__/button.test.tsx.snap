// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button tests renders correctly 1`] = `
<DocumentFragment>
  <button
    class="button small"
    style="background-color: blue;"
  >
    Test Btn
  </button>
</DocumentFragment>
`;
